### PR TITLE
ZEN-29010: Rewrite DNSMonitor to remove reference to Nagios GPL Plugins

### PR DESCRIPTION
**Pull request by olst**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,12 +12,12 @@
...
 NAME = 'ZenPacks.zenoss.DnsMonitor'
-VERSION = '2.1.0'
+VERSION = '3.0.0'
 AUTHOR = 'Zenoss'
 LICENSE = ''
 NAMESPACE_PACKAGES = ['ZenPacks', 'ZenPacks.zenoss']
 PACKAGES = ['ZenPacks', 'ZenPacks.zenoss', 'ZenPacks.zenoss.DnsMonitor']
-INSTALL_REQUIRES = []
+INSTALL_REQUIRES = ['ZenPacks.zenoss.PythonCollector']
 COMPAT_ZENOSS_VERS = '>=2.5.70'
```
Comment (byshovets) [inline setup.py]: Compatible Zenoss version for Python Collector >= 4.2

File: ZenPacks/zenoss/DnsMonitor/datasources/DnsMonitorDataSource.py [hunk 2]
```diff
@@ -29,84 +53,158 @@ class DnsMonitorDataSource(ZenPackPersistence, RRDDataSource.RRDDataSource):
...
+            'message': message,
+            'device': config.id,
+            'eventClass': ds0.eventClass,
+            'severity': ZenEventClasses.Clear
+        })
+
+        return data
+
+    def onError(self, result, config):
+        ds0 = config.datasources[0]
```
Comment (byshovets) [inline ZenPacks/zenoss/DnsMonitor/datasources/DnsMonitorDataSource.py]: What about measuring execution time here as well as in onSuccess?

File: ZenPacks/zenoss/DnsMonitor/datasources/DnsMonitorDataSource.py [hunk 3]
```diff
@@ -29,84 +53,158 @@ class DnsMonitorDataSource(ZenPackPersistence, RRDDataSource.RRDDataSource):
...
+        params = cls.params(datasource, context)
+        return (
+            context.device().id,
+            datasource.getCycleTime(context),
+            datasource.rrdTemplate().id,
+            datasource.id,
+            datasource.plugin_classname,
+            params.get('hostname'),
+            params.get('dnsServer'),
+            params.get('expectedIpAddress'),
```
Comment (byshovets) [inline ZenPacks/zenoss/DnsMonitor/datasources/DnsMonitorDataSource.py]: Looks kind of confusing for me, why do you need all these params in config key?

**Comment by olst** [comment]
Messages from old zenpack:
https://drive.google.com/file/d/1jMuLrlyhFdsSaNSoxz533cCdgDB3B1Re
https://drive.google.com/file/d/1WCwMnRMQ7XNXxGmPmxbkociKX4mC0mAW
https://drive.google.com/file/d/1wuu_iQU0cs0brT36uT1OwGRNjjMw4dAw

**Comment by olst** [comment]
During ZP upgrade all custom DNS datasources are left like they have been created.
Looks like migrate is needed for default DnsMonitor datasource, because it's loaded from objects.xml.

**Comment by olst** [comment]
it's not a ususal migration, looks like it should be like in `migrate` commit.
https://youtu.be/atFvk_jpcDQ

**Comment by olst** [comment]
I've added DnsMonitor datasourse to other template in old version of ZP. Then, I upgraded ZP to a new version. DnsMonitor datasource is not broken and monitoring is ok. Please see this video:
https://youtu.be/_Nkoppvxi8Q

**Pull request merged**